includes/classes/prod_cat_pulldown.php

```diff
@@ -0,0 +1,186 @@
+				'pd.products_description',
+				'p.products_id',
+			];
+
+			$parm_2 = '%2$s ';
+			if (!$this->show_model) {
+				$parm_2 = '';
+			}
+
+			$this->sort = ' ORDER BY pd.products_name';
+			$this->output_string = '%1$s ' . $parm_2 . '(%3$s)';// format string with name first
+
+			if ($this->order_by == 'model') {
+				$this->sort = ' ORDER by p.products_model';
+				$this->output_string = (!empty($parm_2) ? $parm_2 . '-' : '') . ' %1$s (%3$s)'; // format string with model first
+			}
+
+			$this->sql = "SELECT p.products_id as common_id, pd.products_id, p.products_sort_order, p.products_price, p.products_model, ptc.categories_id
+                FROM " . TABLE_PRODUCTS . " p
+                LEFT JOIN " . TABLE_PRODUCTS_DESCRIPTION . " pd USING (products_id)
+                LEFT JOIN " . TABLE_PRODUCTS_TO_CATEGORIES . " ptc USING (products_id) "
+				. $this->attributes_join .
+				" WHERE pd.language_id = " . (int)$_SESSION['languages_id'];
+		}
+
+		private function run_sql()
+		{
+			global $db, $currencies;
+
+			$search = '';
+			if (!empty($this->keywords)) {
+				$search = zen_build_keyword_where_clause($this->keyword_search_fields,
+					trim(zen_db_input(zen_db_prepare_input($this->keywords))));
+			}
+			$this->sql .= $search;
+
+			switch (true) {
+				case ($this->option_filter === -1): // no selection made: do not list any products
+				case ($this->option_filter === ''): // no selection made: do not list any categories
+					// no selection made yet
+					break;
+				case ((int)$this->option_filter > 0) : // an Option Name was selected: show only products using attributes with this Option Name
+					$this->sql .= $this->condition;
+					$this->sql .= $this->sort;
+					break;
+				default: //legacy: show all products with attributes
+					$this->sql .= $this->sort;
+					break;
+			}
+
+			$results = $db->Execute($this->sql);
+
+			foreach ($results as $result) {
+				if (!in_array($result['common_id'], $this->exclude)) {
+					switch (true) {
+						case ($this->pulldown == 'category'):
+							$this->values[] = [
+								'id' => $result['categories_id'],
+								'text' => $this->category_text($result),
+							];
+							break;
+						default:
+							$display_price = zen_get_products_base_price($result['products_id']);
+							$name = zen_get_products_name($result['products_id']);
+							$this->values[] = [
+								'id' => $result['products_id'],
+								'text' => sprintf($this->output_string, trim(zen_clean_html($name)),
+										($this->show_model ? ' [' . $result['products_model'] . '] ' : ''),
+										$currencies->format($display_price)) . ($this->show_id ? ' - ID# ' . $result['products_id'] : ''),
+							];
+							break;
+					}
+				}
+			}
+		}
+
+		private function category_text($category)
+		{
+			if ($this->attributes) {
+				$text = $category['categories_name'];
+				if ($this->show_full_path) {
+					$text = zen_output_generated_category_path($category['categories_id']);
+				}
+			} else {
```

Comment:
Before this closing parentheses, could just return `$text` and then remove the else statement below.

Reply:
this is true; and a question of style.  i'm a fan of minimizing the number of places that a function exits, ie only having 1 `return`.  but again, i'm not opposed to this idea.

Reply:
FWIW I'm a big fan of early-return, so that it's easier to follow into what actual logic is happening after all exclusions are accounted for or no further processing is required.

Reply:
i am not sure if the code is clearer; but i have done the early return as suggested.